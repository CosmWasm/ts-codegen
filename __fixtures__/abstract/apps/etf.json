{
  "contract_name": "module-schema",
  "contract_version": "0.9.0",
  "idl_version": "1.0.0",
  "instantiate": {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "title": "InstantiateMsg",
    "description": "Init msg",
    "type": "object",
    "required": [
      "fee",
      "provider_addr",
      "token_code_id"
    ],
    "properties": {
      "fee": {
        "description": "Fee charged on withdrawal",
        "allOf": [
          {
            "$ref": "#/definitions/Decimal"
          }
        ]
      },
      "provider_addr": {
        "description": "Address of the service provider which receives the fee.",
        "type": "string"
      },
      "token_code_id": {
        "description": "Code-id used to create the LP token",
        "type": "integer",
        "format": "uint64",
        "minimum": 0.0
      },
      "token_name": {
        "description": "Name of the etf token",
        "type": [
          "string",
          "null"
        ]
      },
      "token_symbol": {
        "description": "Symbol of the etf token",
        "type": [
          "string",
          "null"
        ]
      }
    },
    "additionalProperties": false,
    "definitions": {
      "Decimal": {
        "description": "A fixed-point decimal value with 18 fractional digits, i.e. Decimal(1_000_000_000_000_000_000) == 1.0\n\nThe greatest possible value that can be represented is 340282366920938463463.374607431768211455 (which is (2^128 - 1) / 10^18)",
        "type": "string"
      }
    }
  },
  "execute": {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "title": "ExecuteMsg",
    "oneOf": [
      {
        "description": "Provide liquidity to the attached proxy using a native token.",
        "type": "object",
        "required": [
          "provide_liquidity"
        ],
        "properties": {
          "provide_liquidity": {
            "type": "object",
            "required": [
              "asset"
            ],
            "properties": {
              "asset": {
                "$ref": "#/definitions/AssetBase_for_String"
              }
            },
            "additionalProperties": false
          }
        },
        "additionalProperties": false
      },
      {
        "description": "Set the withdraw fee",
        "type": "object",
        "required": [
          "set_fee"
        ],
        "properties": {
          "set_fee": {
            "type": "object",
            "required": [
              "fee"
            ],
            "properties": {
              "fee": {
                "$ref": "#/definitions/Decimal"
              }
            },
            "additionalProperties": false
          }
        },
        "additionalProperties": false
      }
    ],
    "definitions": {
      "AssetBase_for_String": {
        "description": "Represents a fungible asset with a known amount\n\nEach asset instance contains two values: `info`, which specifies the asset's type (CW20 or native), and its `amount`, which specifies the asset's amount.",
        "type": "object",
        "required": [
          "amount",
          "info"
        ],
        "properties": {
          "amount": {
            "description": "Specifies the asset's amount",
            "allOf": [
              {
                "$ref": "#/definitions/Uint128"
              }
            ]
          },
          "info": {
            "description": "Specifies the asset's type (CW20 or native)",
            "allOf": [
              {
                "$ref": "#/definitions/AssetInfoBase_for_String"
              }
            ]
          }
        },
        "additionalProperties": false
      },
      "AssetInfoBase_for_String": {
        "description": "Represents the type of an fungible asset.\n\nEach **asset info** instance can be one of three variants:\n\n- Native SDK coins. To create an **asset info** instance of this type, provide the denomination. - CW20 tokens. To create an **asset info** instance of this type, provide the contract address.",
        "oneOf": [
          {
            "type": "object",
            "required": [
              "native"
            ],
            "properties": {
              "native": {
                "type": "string"
              }
            },
            "additionalProperties": false
          },
          {
            "type": "object",
            "required": [
              "cw20"
            ],
            "properties": {
              "cw20": {
                "type": "string"
              }
            },
            "additionalProperties": false
          }
        ]
      },
      "Decimal": {
        "description": "A fixed-point decimal value with 18 fractional digits, i.e. Decimal(1_000_000_000_000_000_000) == 1.0\n\nThe greatest possible value that can be represented is 340282366920938463463.374607431768211455 (which is (2^128 - 1) / 10^18)",
        "type": "string"
      },
      "Uint128": {
        "description": "A thin wrapper around u128 that is using strings for JSON encoding/decoding, such that the full u128 range can be used for clients that convert JSON numbers to floats, like JavaScript and jq.\n\n# Examples\n\nUse `from` to create instances of this and `u128` to get the value out:\n\n``` # use cosmwasm_std::Uint128; let a = Uint128::from(123u128); assert_eq!(a.u128(), 123);\n\nlet b = Uint128::from(42u64); assert_eq!(b.u128(), 42);\n\nlet c = Uint128::from(70u32); assert_eq!(c.u128(), 70); ```",
        "type": "string"
      }
    }
  },
  "query": {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "title": "QueryMsg",
    "oneOf": [
      {
        "description": "Returns [`StateResponse`]",
        "type": "object",
        "required": [
          "state"
        ],
        "properties": {
          "state": {
            "type": "object",
            "additionalProperties": false
          }
        },
        "additionalProperties": false
      }
    ]
  },
  "migrate": {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "title": "MigrateMsg",
    "description": "An empty struct that serves as a placeholder in different places, such as contracts that don't set a custom message.\n\nIt is designed to be expressable in correct JSON and JSON Schema but contains no meaningful data. Previously we used enums without cases, but those cannot represented as valid JSON Schema (https://github.com/CosmWasm/cosmwasm/issues/451)",
    "type": "object"
  },
  "sudo": null,
  "responses": {
    "state": {
      "$schema": "http://json-schema.org/draft-07/schema#",
      "title": "StateResponse",
      "type": "object",
      "required": [
        "fee",
        "liquidity_token"
      ],
      "properties": {
        "fee": {
          "$ref": "#/definitions/Decimal"
        },
        "liquidity_token": {
          "type": "string"
        }
      },
      "additionalProperties": false,
      "definitions": {
        "Decimal": {
          "description": "A fixed-point decimal value with 18 fractional digits, i.e. Decimal(1_000_000_000_000_000_000) == 1.0\n\nThe greatest possible value that can be represented is 340282366920938463463.374607431768211455 (which is (2^128 - 1) / 10^18)",
          "type": "string"
        }
      }
    }
  }
}
