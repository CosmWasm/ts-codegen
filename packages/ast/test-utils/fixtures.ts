// @ts-nocheck
import { JSONSchema } from '../src/types';

import canExecuteRelayResponseData from '../../../__fixtures__/vectis/govec/can_execute_relay_response.json';
import cosmosMsgForEmptyData from '../../../__fixtures__/vectis/govec/cosmos_msg_for__empty.json';
import executeMsgForEmptyData from '../../../__fixtures__/vectis/govec/execute_msg_for__empty.json';
import infoResponseData from '../../../__fixtures__/vectis/govec/info_response.json';
import relayTransactionData from '../../../__fixtures__/vectis/govec/relay_transaction.json';

import nftIdlVersionData from '../../../__fixtures__/idl-version/accounts-nft/account-nft.json';

export const nftIdlVersion: JSONSchema = nftIdlVersionData;
export const vectisCanExecuteRelayResponse: JSONSchema = canExecuteRelayResponseData;
export const vectisCosmosMsgForEmpty: JSONSchema = cosmosMsgForEmptyData;
export const vectisExecuteMsgForEmpty: JSONSchema = executeMsgForEmptyData;
export const vectisInfoResponse: JSONSchema = infoResponseData;
export const vectisRelayTransaction: JSONSchema = relayTransactionData;
