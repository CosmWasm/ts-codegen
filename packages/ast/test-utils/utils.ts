import generate from '@babel/generator';
import { RenderOptions } from '@cosmwasm/ts-codegen-ast/types';
import { ExecuteMsg, IDLObject, JSONSchema, QueryMsg } from '@cosmwasm/ts-codegen-types';
import { readFileSync } from 'fs';
import { sync as glob } from 'glob';
import { join } from 'path';

import { RenderContext } from '../src/context';

export const expectCode = (ast: any): void => {
  expect(
    generate(ast).code
  ).toMatchSnapshot();
};

export const printCode = (ast: any): void => {
  console.log(
    generate(ast).code
  );
};

export const makeContext = (
  schema: JSONSchema,
  options?: RenderOptions,
  responses?: Record<string, JSONSchema>
) => {
  return new RenderContext({
    schemas: [schema],
    responses
  }, options);
};

interface GlobContractLegacy {
  name: `/${string}.json`;
  content: JSONSchema;
}

interface GlobContract {
  name: `/${string}.json`;
  content: IDLObject;
}

const fixtureDir = join(__dirname, '../../../__fixtures__');
const globCache: Record<string, GlobContractLegacy[] | GlobContract[]> = {};

export const globIdlBasedContracts = (p: string): GlobContract[] => {
  if (globCache[p]) return globCache[p] as GlobContract[];
  // @ts-ignore
  const contracts: GlobContract[] = glob(join(fixtureDir, p, '/*.json')).map(file => {
    return {
      name: file.split(join('__fixtures__', p))[1],
      content: JSON.parse(readFileSync(file, 'utf-8'))
    };
  });
  globCache[p] = contracts;
  return contracts;
};

export const globLegacyContracts = (p: string): GlobContractLegacy[] => {
  if (globCache[p]) return globCache[p] as GlobContractLegacy[];
  // @ts-ignore
  const contracts: GlobContractLegacy[] = glob(join(fixtureDir, p, '/*.json')).map(file => {
    return {
      name: file.split(join('__fixtures__', p))[1],
      content: JSON.parse(readFileSync(file, 'utf-8'))
    };
  });
  globCache[p] = contracts;
  return contracts;
};

export const getMsgExecuteLegacyFixture = (
  p: string,
  name: `/${string}.json`
): ExecuteMsg => {
  const contracts = globLegacyContracts(p);
  const contract = contracts.find(a => a.name === name);
  return contract.content as ExecuteMsg;
};

export const getMsgQueryLegacyFixture = (
  p: string,
  name: `/${string}.json`
): QueryMsg => {
  const contracts = globLegacyContracts(p);
  const contract = contracts.find(a => a.name === name);
  return contract.content as QueryMsg;
};

export const getLegacyFixture = (
  p: string,
  name: `/${string}.json`
): JSONSchema => {
  const contracts = globLegacyContracts(p);
  const contract = contracts.find(a => a.name === name);
  return contract.content as JSONSchema;
};
