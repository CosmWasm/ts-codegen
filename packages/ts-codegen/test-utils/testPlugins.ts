import {
  BuilderContext,
  ContractInfo,
  RenderContext,
  RenderContextBase,
  RenderOptions,
} from "wasm-ast-types";
import { ClientPlugin, MessageComposerPlugin, TSBuilderOptions } from "../src";
import { ContractsContextProviderPlugin } from "../src/plugins/provider";
import { ContractsProviderBundlePlugin } from "../src/plugins/provider-bundle";

export class TestClientPlugin extends ClientPlugin {
  builderContext: BuilderContext;

  constructor(opts: RenderOptions, builderContext: BuilderContext) {
    super(opts);
    this.builderContext = builderContext;
  }

  initContext(
    contract: ContractInfo,
    options?: RenderOptions
  ): RenderContextBase<RenderOptions> {
    return new RenderContext(contract, options, this.builderContext);
  }
}

export class TestMessageComposerPlugin extends MessageComposerPlugin {
  builderContext: BuilderContext;

  constructor(opts: RenderOptions, builderContext: BuilderContext) {
    super(opts);
    this.builderContext = builderContext;
  }

  initContext(
    contract: ContractInfo,
    options?: RenderOptions
  ): RenderContextBase<RenderOptions> {
    return new RenderContext(contract, options, this.builderContext);
  }
}

export class TestContractsContextProviderPlugin extends ContractsContextProviderPlugin {
  builderContext: BuilderContext;

  constructor(opts: TSBuilderOptions, builderContext: BuilderContext) {
    super(opts);
    this.builderContext = builderContext;
  }

  initContext(
    contract: ContractInfo,
    options?: TSBuilderOptions
  ): RenderContextBase<TSBuilderOptions> {
    return new RenderContext(contract, options, this.builderContext);
  }
}

export class TestContractsProviderBundlePlugin extends ContractsProviderBundlePlugin {
  builderContext: BuilderContext;

  constructor(opts: TSBuilderOptions, builderContext: BuilderContext) {
    super(opts);
    this.builderContext = builderContext;
  }

  initContext(
    contract: ContractInfo,
    options?: TSBuilderOptions
  ): RenderContextBase<TSBuilderOptions> {
    return new RenderContext(contract, options, this.builderContext);
  }
}
