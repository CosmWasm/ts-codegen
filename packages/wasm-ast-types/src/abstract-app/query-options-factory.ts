import { RenderContext } from '../context';
import * as t from '@babel/types';
import {
  createExtractTypeAnnotation,
  createTypedObjectParams,
  getMessageProperties,
  getResponseType,
  identifier,
  shorthandProperty,
  tsObjectPattern
} from '../utils';
import { camel } from 'case';
import { ExecuteMsg, QueryMsg } from '../types';
import { getWasmMethodArgs } from '../client';

type GenerationType = 'abstract-app' | 'contract';

/**
 * Create a react-query factory for use in queries.
 * @param context
 * @param moduleName
 * @param queryMsg
 * @param isAbstractApp
 * @todo get rid of `isAbtsractApp` in favor of context option
 */
export function createQueryOptionsFactory(
  context: RenderContext,
  moduleName: string,
  queryMsg: QueryMsg,
  generationType: GenerationType
) {
  context.addUtil('createQueryKeys');

  const queryKeys = getMessageProperties(queryMsg).map((schema) => {
    return createQueryOptions(context, schema, generationType);
  });

  return t.exportNamedDeclaration(
    t.variableDeclaration('const', [
      t.variableDeclarator(
        t.identifier(`${camel(moduleName)}Queries`),
        t.callExpression(t.identifier('createQueryKeys'), [
          t.stringLiteral(moduleName),
          t.objectExpression(queryKeys)
        ])
      )
    ])
  );
}

/**
 * Query keys for Abstract app modules
 */
const ABSTRACT_APP_QUERY_KEYS = t.objectExpression([
  // Abstract account Id
  t.objectProperty(
    t.identifier('accountId'),
    t.memberExpression(
      t.memberExpression(
        t.identifier('queryClient'),
        t.identifier('accountQueryClient'),
        false
      ),
      t.identifier('accountId'),
      false
    )
  ),
  // Abstract module Id
  t.objectProperty(
    t.identifier('moduleId'),
    t.memberExpression(
      t.identifier('queryClient'),
      t.identifier('moduleId'),
      false
    )
  )
]);

/**
 * Query keys for Abstract app modules
 */
const CONTRACT_QUERY_CLIENT_KEYS = t.objectExpression([
  // Contract address
  t.objectProperty(
    t.identifier('address'),
    t.memberExpression(
      t.identifier('queryClient'),
      t.identifier('contractAddress'),
      false
    )
  )
]);

/**
 * Generate the query options for the query client.
 * @param context
 * @param jsonschema
 * @param isAbstractApp
 * TODO: get rid of isAbstractApp in favor of context option
 */
const createQueryOptions = (
  context: RenderContext,
  jsonschema: any,
  generationType: GenerationType
) => {
  const underscoreName = Object.keys(jsonschema.properties)[0];
  const methodName = camel(underscoreName);

  let param = createTypedObjectParams(
    context,
    jsonschema.properties[underscoreName]
  );

  // TODO: this is a hack to get the type annotation to work
  // all type annotations in the future should be the extracted and camelized type
  if (
    param &&
    param.typeAnnotation.type === 'TSTypeAnnotation' &&
    param.typeAnnotation.typeAnnotation.type === 'TSTypeLiteral'
  ) {
    param = t.identifier('params');
    param.typeAnnotation = createExtractTypeAnnotation(
      underscoreName,
      'QueryMsg'
    );
  }

  return t.objectProperty(
    t.identifier(methodName),
    t.arrowFunctionExpression(
      [
        identifier(
          'queryClient',
          t.tsTypeAnnotation(
            t.tsTypeReference(t.identifier('AutocompounderQueryClient'))
          )
        ),
        ...(param ? [param] : [])
      ],
      t.objectExpression([
        t.objectProperty(
          t.identifier('queryKey'),
          t.arrayExpression([
            generationType === 'abstract-app'
              ? ABSTRACT_APP_QUERY_KEYS
              : CONTRACT_QUERY_CLIENT_KEYS,
            ...(param ? [t.identifier('params')] : [])
          ])
        ),
        t.objectProperty(
          t.identifier('queryFn'),
          t.arrowFunctionExpression(
            [t.identifier('ctx')],
            t.callExpression(
              t.memberExpression(
                t.identifier('queryClient'),
                t.identifier(methodName),
                false
              ),
              param ? [t.identifier('params')] : []
            )
          )
        )
      ])
    )
  );
};
