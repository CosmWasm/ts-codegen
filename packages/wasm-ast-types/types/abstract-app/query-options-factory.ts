import { RenderContext } from '../context';
import * as t from '@babel/types';
import { QueryMsg } from '../types';

/**
 * Create a react-query factory for use in queries.
 * @param context
 * @param moduleName
 * @param queryMsg
 * @todo get rid of `isAbtsractApp` in favor of context option
 */
export declare const createQueryOptionsFactory: (
  context: RenderContext,
  moduleName: string,
  queryMsg: QueryMsg,
  isAbstractApp: boolean
) => t.ExportNamedDeclaration;
