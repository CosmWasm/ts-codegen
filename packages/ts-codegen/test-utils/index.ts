import {
  BuilderContext,
  ContractInfo,
  ReactQueryOptions,
  RenderOptions,
  TSTypesOptions,
} from "wasm-ast-types";
import { ReactQueryPlugin } from "../src/plugins/react-query";
import { TypesPlugin } from "../src/plugins/types";
import { ClientPlugin } from "../src/plugins/client";
import { MessageComposerPlugin } from "../src/plugins/message-composer";
import { RecoilPlugin } from "../src/plugins/recoil";
import { MessageBuilderPlugin } from "../src/plugins/message-builder";
import { ContractsContextProviderPlugin } from "../src/plugins/provider";
import { ContractsProviderBundlePlugin } from "../src/plugins/provider-bundle";
import { TSBuilderOptions } from "../src/builder";
import { createDefaultContractInfo } from "../src/utils/contracts";
import {
  TestContractsContextProviderPlugin,
  TestClientPlugin,
  TestMessageComposerPlugin,
  TestContractsProviderBundlePlugin,
} from "./testPlugins";
import deepmerge from "deepmerge";

export const testDefaultOptions: RenderOptions = {
  enabled: true,
  types: {
    enabled: true,
    itemsUseTuples: false,
    aliasExecuteMsg: false,
  },
  client: {
    enabled: true,
    execExtendsQuery: true,
    noImplicitOverride: false,
  },
  recoil: {
    enabled: true,
  },
  messageComposer: {
    enabled: true,
  },
  messageBuilder: {
    enabled: true,
  },
  reactQuery: {
    enabled: true,
    optionalClient: false,
    version: "v4",
    mutations: false,
    camelize: true,
    queryKeys: false,
  },
  useContractsHook: {
    enabled: true,
  },
};

export async function generateReactQuery(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  opts?: ReactQueryOptions
) {
  const options = opts
    ? deepmerge(testDefaultOptions, { reactQuery: opts })
    : testDefaultOptions;

  await new ReactQueryPlugin(options).render(
    outPath,
    contractName,
    contractInfo
  );
}

export async function generateTypes(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  opts?: TSTypesOptions
) {
  await new TypesPlugin(opts ?? testDefaultOptions).render(
    outPath,
    contractName,
    contractInfo
  );
}

export async function generateClient(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  builderContext?: BuilderContext,
  opts?: RenderOptions
) {
  if (builderContext) {
    await new TestClientPlugin(
      opts ?? testDefaultOptions,
      builderContext
    ).render(outPath, contractName, contractInfo);
  } else {
    await new ClientPlugin(opts ?? testDefaultOptions).render(
      outPath,
      contractName,
      contractInfo
    );
  }
}

export async function generateMessageComposer(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  builderContext?: BuilderContext,
  opts?: RenderOptions
) {
  if (builderContext) {
    await new TestMessageComposerPlugin(
      opts ?? testDefaultOptions,
      builderContext
    ).render(outPath, contractName, contractInfo);
  } else {
    await new MessageComposerPlugin(opts ?? testDefaultOptions).render(
      outPath,
      contractName,
      contractInfo
    );
  }
}

export async function generateMessageBuilder(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  opts?: RenderOptions
) {
  await new MessageBuilderPlugin(opts ?? testDefaultOptions).render(
    outPath,
    contractName,
    contractInfo
  );
}

export async function generateRecoil(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  opts?: RenderOptions
) {
  await new RecoilPlugin(opts ?? testDefaultOptions).render(
    outPath,
    contractName,
    contractInfo
  );
}

export async function generateContractHooks(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  builderContext?: BuilderContext
) {
  await new TestContractsContextProviderPlugin(
    {
      enabled: true,
      useShorthandCtor: true,
      useContractsHook: {
        enabled: true,
      },
    },
    builderContext
  ).render(outPath, contractName, contractInfo);

  await new TestContractsProviderBundlePlugin(
    {
      enabled: true,
      useShorthandCtor: true,
      useContractsHook: {
        enabled: true,
      },
    },
    builderContext
  ).render(outPath, undefined, contractInfo ?? createDefaultContractInfo());
}
