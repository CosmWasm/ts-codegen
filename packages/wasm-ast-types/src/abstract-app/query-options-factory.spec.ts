import { expectCode, makeContext } from '../../test-utils';
import { query as etfQuery } from '../../../../__fixtures__/abstract/apps/etf.json';
import { query as subscriptionQuery } from '../../../../__fixtures__/abstract/apps/subscription.json';
import { query as autocompounderQuery } from '../../../../__fixtures__/abstract/apps/autocompounder.json';
import { createQueryOptionsFactory } from './query-options-factory';
import query_msg from '../../../../__fixtures__/basic/query_msg.json';

it('etf', () => {
  const ctx = makeContext(etfQuery);
  expectCode(createQueryOptionsFactory(ctx, 'Etf', etfQuery, 'abstract-app'));
});

it('subscription', () => {
  const ctx = makeContext(subscriptionQuery);
  expectCode(
    createQueryOptionsFactory(ctx, 'Subscription', subscriptionQuery, 'abstract-app')
  );
});

it('queryMsg', () => {
  const ctx = makeContext(query_msg);
  expectCode(createQueryOptionsFactory(ctx, 'Contract', query_msg, 'contract'));
});

it('autocompounderQueryMsg', () => {
  const ctx = makeContext(autocompounderQuery);
  expectCode(
    createQueryOptionsFactory(
      ctx,
      'Autocompounder',
      autocompounderQuery,
      'contract'
    )
  );
});
