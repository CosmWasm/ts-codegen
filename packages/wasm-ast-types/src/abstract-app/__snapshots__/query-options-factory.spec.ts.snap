// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`autocompounderQueryMsg 1`] = `
"export const autocompounderQueries = createQueryKeys(\\"Autocompounder\\", {
  config: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }],
    queryFn: ctx => queryClient.config()
  }),
  pendingClaims: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    pending_claims: unknown;
  }>[\\"pending_claims\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.pendingClaims(params)
  }),
  claims: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    claims: unknown;
  }>[\\"claims\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.claims(params)
  }),
  allClaims: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    all_claims: unknown;
  }>[\\"all_claims\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.allClaims(params)
  }),
  latestUnbonding: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }],
    queryFn: ctx => queryClient.latestUnbonding()
  }),
  totalLpPosition: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }],
    queryFn: ctx => queryClient.totalLpPosition()
  }),
  balance: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    balance: unknown;
  }>[\\"balance\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.balance(params)
  })
});"
`;

exports[`etf 1`] = `
"export const etfQueries = createQueryKeys(\\"Etf\\", {
  state: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      accountId: queryClient.accountQueryClient.accountId,
      moduleId: queryClient.moduleId
    }],
    queryFn: ctx => queryClient.state()
  })
});"
`;

exports[`queryMsg 1`] = `
"export const contractQueries = createQueryKeys(\\"Contract\\", {
  ownerOf: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    owner_of: unknown;
  }>[\\"owner_of\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.ownerOf(params)
  }),
  approval: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    approval: unknown;
  }>[\\"approval\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.approval(params)
  }),
  approvals: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    approvals: unknown;
  }>[\\"approvals\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.approvals(params)
  }),
  allOperators: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    all_operators: unknown;
  }>[\\"all_operators\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.allOperators(params)
  }),
  numTokens: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }],
    queryFn: ctx => queryClient.numTokens()
  }),
  contractInfo: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }],
    queryFn: ctx => queryClient.contractInfo()
  }),
  nftInfo: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    nft_info: unknown;
  }>[\\"nft_info\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.nftInfo(params)
  }),
  allNftInfo: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    all_nft_info: unknown;
  }>[\\"all_nft_info\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.allNftInfo(params)
  }),
  tokens: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    tokens: unknown;
  }>[\\"tokens\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.tokens(params)
  }),
  allTokens: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    all_tokens: unknown;
  }>[\\"all_tokens\\"]>) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }, params],
    queryFn: ctx => queryClient.allTokens(params)
  }),
  minter: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }],
    queryFn: ctx => queryClient.minter()
  }),
  collectionInfo: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      address: queryClient.contractAddress
    }],
    queryFn: ctx => queryClient.collectionInfo()
  })
});"
`;

exports[`subscription 1`] = `
"export const subscriptionQueries = createQueryKeys(\\"Subscription\\", {
  state: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      accountId: queryClient.accountQueryClient.accountId,
      moduleId: queryClient.moduleId
    }],
    queryFn: ctx => queryClient.state()
  }),
  config: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      accountId: queryClient.accountQueryClient.accountId,
      moduleId: queryClient.moduleId
    }],
    queryFn: ctx => queryClient.config()
  }),
  fee: (queryClient: AutocompounderQueryClient) => ({
    queryKey: [{
      accountId: queryClient.accountQueryClient.accountId,
      moduleId: queryClient.moduleId
    }],
    queryFn: ctx => queryClient.fee()
  }),
  subscriberState: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    subscriber_state: unknown;
  }>[\\"subscriber_state\\"]>) => ({
    queryKey: [{
      accountId: queryClient.accountQueryClient.accountId,
      moduleId: queryClient.moduleId
    }, params],
    queryFn: ctx => queryClient.subscriberState(params)
  }),
  contributorState: (queryClient: AutocompounderQueryClient, params: CamelCasedProperties<Extract<QueryMsg, {
    contributor_state: unknown;
  }>[\\"contributor_state\\"]>) => ({
    queryKey: [{
      accountId: queryClient.accountQueryClient.accountId,
      moduleId: queryClient.moduleId
    }, params],
    queryFn: ctx => queryClient.contributorState(params)
  })
});"
`;
